### ISSUE TITLE
Recognize Symbols Defined by areduce

### ISSUE THREAD

**Issue by alexanderkiel** [cursive-ide/cursive#435]
The macro [areduce](http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/areduce) defines two symbols specified by the arguments `ret` and `init`. Symbols placed there are currently not recognized by Cursive.

My use case is the following function:

``` clojure
(defn decode-len
  "Returns the length represented by the encoded bytes."
  [^bytes b]
  {:pre [(= 4 (count b))]}
  (areduce b i ret 0
           (+ ret (bit-shift-left (bit-and 0xff (int (aget b i))) (* i 8)))))
```

here `i` and `ret` are marked as `cannot be resolved` by v0.1.29.

**Comment by trimtab613** [comment]
I think this ticket can be generalized to any macro that defines things for access in its body.

**Comment by cursiveclojure** [comment]
Yes, see #147, which is the catch-all for this problem. That said, that's more oriented to third-party macros, core ones like areduce I normally add to Cursive directly.

**Comment by conan** [comment]
+1
